eks/modules/eks/cluster-iam.tf

```diff
@@ -0,0 +1,27 @@
+
+#  * IAM Role to allow EKS service to manage other AWS services
+data "aws_iam_policy_document" "cluster-role-policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["eks.amazonaws.com"]
+    }
+  }
+}
+resource "aws_iam_role" "aws-cluster" {
+  name = "terraform-eks-aws-cluster"
```

Comment:
`name = "${var.cluster-name}-cluster"`
needs to be set through variable interpolation otherwise, the deployment will be only possible for one cluster setup.